### ISSUE TITLE
Why passing non-zero timeout value to `wait_for_pending_acks_returns` causes immediate timeout exception?

### ISSUE THREAD

**Issue by fsalehpour** [php-amqplib/php-amqplib#462]
I run [amqp_publisher_with_confirms_mandatory.php](https://github.com/php-amqplib/php-amqplib/blob/master/demo/amqp_publisher_with_confirms_mandatory.php) demo without any problems.

But as soon as I pass any value other than `0` for `timeout` to `wait_for_pending_acks_returns` on line 58 I get (e.g. with `timeout` value of `1`):

```
PHP Fatal error:  Uncaught PhpAmqpLib\Exception\AMQPTimeoutException: The connection timed out after 1 sec while awaiting incoming data in /Home/code/php-amqplib/PhpAmqpLib/Wire/AMQPReader.php:132
Stack trace:
#0 /Home/code/php-amqplib/PhpAmqpLib/Wire/AMQPReader.php(148): PhpAmqpLib\Wire\AMQPReader->wait()
#1 /Home/code/php-amqplib/PhpAmqpLib/Wire/AMQPReader.php(106): PhpAmqpLib\Wire\AMQPReader->rawread(7)
#2 /Home/code/php-amqplib/PhpAmqpLib/Connection/AbstractConnection.php(508): PhpAmqpLib\Wire\AMQPReader->read(7)
#3 /Home/code/php-amqplib/PhpAmqpLib/Connection/AbstractConnection.php(555): PhpAmqpLib\Connection\AbstractConnection->wait_frame(1)
#4 /Home/code/php-amqplib/PhpAmqpLib/Channel/AbstractChannel.php(217): PhpAmqpLib\Connection\AbstractConnection->wait_channel(1, 1)
#5 /Home/code/php-amqplib/PhpAmqpLib/Channel/AbstractChannel.php(328): PhpAmqpLib\Channel\AbstractChannel->next_frame(1)
#6 /Home/code/php-amqplib/PhpAmqpLib/Channel/AMQPChan in /Home/code/php-amqplib/PhpAmqpLib/Wire/AMQPReader.php on line 132

Fatal error: Uncaught PhpAmqpLib\Exception\AMQPTimeoutException: The connection timed out after 1 sec while awaiting incoming data in /Home/code/php-amqplib/PhpAmqpLib/Wire/AMQPReader.php:132
Stack trace:
#0 /Home/code/php-amqplib/PhpAmqpLib/Wire/AMQPReader.php(148): PhpAmqpLib\Wire\AMQPReader->wait()
#1 /Home/code/php-amqplib/PhpAmqpLib/Wire/AMQPReader.php(106): PhpAmqpLib\Wire\AMQPReader->rawread(7)
#2 /Home/code/php-amqplib/PhpAmqpLib/Connection/AbstractConnection.php(508): PhpAmqpLib\Wire\AMQPReader->read(7)
#3 /Home/code/php-amqplib/PhpAmqpLib/Connection/AbstractConnection.php(555): PhpAmqpLib\Connection\AbstractConnection->wait_frame(1)
#4 /Home/code/php-amqplib/PhpAmqpLib/Channel/AbstractChannel.php(217): PhpAmqpLib\Connection\AbstractConnection->wait_channel(1, 1)
#5 /Home/code/php-amqplib/PhpAmqpLib/Channel/AbstractChannel.php(328): PhpAmqpLib\Channel\AbstractChannel->next_frame(1)
#6 /Home/code/php-amqplib/PhpAmqpLib/Channel/AMQPChan in /Home/code/php-amqplib/PhpAmqpLib/Wire/AMQPReader.php on line 132
```
I was expecting to receive this exception if the RMQ server didn't respond in 1 second. But from what I get I know that it responds immediately.

Could you please help me to figure out what's going on?   
Thank you

**Comment by hairyhum** [comment]
Addressed by reverting #460

**Comment by michaelklishin** [comment]
Thank you, @hairyhum.
